lib/mix/test/mix/tasks/compile.xref_test.exs

```diff
@@ -16,81 +16,81 @@ defmodule Mix.Tasks.Compile.XrefTest do
 
       [xref_manifest] = Mix.Tasks.Compile.Xref.manifests()
 
-      assert_no_warn fn ->
+      assert_no_warn(fn ->
         assert Mix.Tasks.Compile.Xref.run([]) == {:noop, []}
-        refute File.exists? xref_manifest
-      end
+        refute File.exists?(xref_manifest)
+      end)
     end
   end
 
   test "doesn't xref if not stale, unless forced" do
     in_fixture "no_mixfile", fn ->
       write_no_func()
 
-      assert_warn_no_func fn ->
+      assert_warn_no_func(fn ->
         assert Mix.Tasks.Compile.Elixir.run([]) == {:ok, []}
         assert {:noop, [_]} = Mix.Tasks.Compile.Xref.run([])
-      end
+      end)
 
-      assert_no_warn fn ->
-        assert {:noop, [_]} = Mix.Tasks.Compile.Xref.run([])
-      end
+      assert_no_warn(fn -> assert {:noop, [_]} = Mix.Tasks.Compile.Xref.run([]) end)
 
       Mix.Task.reenable("xref")
 
-      assert_warn_no_func fn ->
-        assert {:noop, [_]} = Mix.Tasks.Compile.Xref.run(["--force"])
-      end
+      assert_warn_no_func(fn -> assert {:noop, [_]} = Mix.Tasks.Compile.Xref.run(["--force"]) end)
     end
   end
 
   test "xrefs if stale" do
     in_fixture "no_mixfile", fn ->
       write_no_func()
 
-      assert_warn_no_func fn ->
+      assert_warn_no_func(fn ->
         assert Mix.Tasks.Compile.Elixir.run([]) == {:ok, []}
         file = Path.absname("lib/a.ex")
-        assert {:noop, [%Mix.Task.Compiler.Diagnostic{
-          compiler_name: "Xref",
-          file: ^file,
-          message: "function B.no_func/0 is undefined or private",
-          position: 2,
-          severity: :warning
-        }]} = Mix.Tasks.Compile.Xref.run([])
-      end
+
+        assert {:noop, [diagnostic]} = Mix.Tasks.Compile.Xref.run([])
+
+        assert %Mix.Task.Compiler.Diagnostic{
+                 compiler_name: "Xref",
+                 file: ^file,
+                 message: "function B.no_func/0 is undefined or private",
+                 position: 2,
+                 severity: :warning
+               } = diagnostic
+      end)
 
       [manifest] = Mix.Tasks.Compile.Elixir.manifests()
       future = {{2020, 1, 1}, {0, 0, 0}}
       File.touch!(manifest, future)
 
       Mix.Task.reenable("xref")
 
-      assert_warn_no_func fn ->
-        assert {:noop, [_]} = Mix.Tasks.Compile.Xref.run([])
-      end
+      assert_warn_no_func(fn -> assert {:noop, [_]} = Mix.Tasks.Compile.Xref.run([]) end)
     end
   end
 
   test "exits if warnings-as-errors" do
     in_fixture "no_mixfile", fn ->
       write_no_func()
 
-      assert_warn_no_func fn ->
+      assert_warn_no_func(fn ->
         assert Mix.Tasks.Compile.Elixir.run([]) == {:ok, []}
-        assert {:error, [%Mix.Task.Compiler.Diagnostic{
+
+        diagnostic = %Mix.Task.Compiler.Diagnostic{
           severity: :error
-        }]} = Mix.Tasks.Compile.Xref.run(["--warnings-as-errors"])
-      end
+        }
+
+        assert {:error, [diagnostic]} = Mix.Tasks.Compile.Xref.run(["--warnings-as-errors"])
```

Comment:
This will overwrite the `diagnostic` variable. Either match with `^diagnostic` or test the variable contents after the assertion.